packages/cosmic-proto/src/address-hooks.js

```diff
@@ -204,10 +222,21 @@ export const splitHookedAddressUnsafe = (
   const { prefix, bytes } = decodeBech32(specimen, charLimit);
 
   const magicLength = ADDRESS_HOOK_MAGIC.length;
+  let version = 0xff;
   for (let i = 0; i < magicLength; i += 1) {
-    if (bytes[i] !== ADDRESS_HOOK_MAGIC[i]) {
-      return { baseAddress: specimen, hookData: new Uint8Array() };
+    let maybeMagicByte = bytes[i];
+    if (i === magicLength - 1) {
+      // Final byte has a low version nibble and a high magic nibble.
+      version = maybeMagicByte & 0x0f;
+      maybeMagicByte &= 0xf0;
     }
+    if (maybeMagicByte !== ADDRESS_HOOK_MAGIC[i]) {
+      return harden({ baseAddress: specimen, hookData: new Uint8Array() });
+    }
+  }
+
+  if (version !== ADDRESS_HOOK_VERSION) {
+    return `Unsupported address hook version ${version}`;
```

Comment:
I didn't get to this function before, but it's bizarre for it to return strings rather than throw exceptions. Please document the reason (or just eliminate the need by e.g. merging it into `splitHookedAddress`).

...and even if it is important to return rather than throw, consider future-proofing by returning a composite `{ baseAddress, hookData } | { error }` record.

Reply:
> it's bizarre

Indeed.  It's the style that the `bech32` package was written in, and it rubbed off on me. :P  Time to clean up that muck.